src/shared/components/GenericList/SortModalPanel.js

```diff
@@ -41,44 +66,66 @@ export const SortModalPanel = ({ sortBy, sort, setSort, disabled = false }) => {
       ]}
       modalOpeningComponent={sortOpeningComponent}
     >
-      <p style={{ padding: '10px' }}>{t('common.sorting.sort-order')}</p>
-      <FlexBox direction="Column">
-        <RadioButton
-          name="sortOrder"
-          value="ASC"
-          checked={order === 'ASC'}
-          text={t('common.sorting.asc')}
-          onChange={event => setOrder(event.target.value)}
-        />
-        <RadioButton
-          name="sortOrder"
-          value="DESC"
-          checked={order === 'DESC'}
-          text={t('common.sorting.desc')}
-          onChange={event => setOrder(event.target.value)}
-        />
-      </FlexBox>
-      <p style={{ padding: '10px' }}>{t('common.sorting.sort-by')}</p>
-      {sortBy && (
-        <FlexBox direction="Column">
-          {Object.entries(sortBy).flatMap(([value]) => {
-            return (
-              <RadioButton
-                name="sortBy"
-                value={value}
-                key={value}
-                checked={name === value}
-                text={
-                  i18n.exists(`common.sorting.${value}`)
-                    ? t(`common.sorting.${value}`)
-                    : value
-                }
-                onChange={event => setName(event.target.value)}
-              />
-            );
-          })}
-        </FlexBox>
-      )}
+      <List
+        separators="All"
+        onItemClick={e => {
+          setOrder(e?.detail?.item?.children[0]?.value);
+        }}
+        accessibleName="sortOrderList"
+      >
+        <GroupHeaderListItem>
+          {t('common.sorting.sort-order')}
+        </GroupHeaderListItem>
+        <CustomListItem>
+          <RadioButton
+            name="sortOrder"
+            value="ASC"
+            checked={order === 'ASC'}
+            onChange={event => setOrder(event.target.value)}
+          />
+          <Text>{t('common.sorting.asc')}</Text>
```

Comment:
why don't you use the text-property of RadioButton anymore?

Reply:
having the text property on radio button causes the border to appear on click also around the text 
![Screenshot 2024-06-19 at 10 36 51](https://github.com/kyma-project/busola/assets/72342415/90d16b04-617a-432d-8973-03384e570c16)